### PR TITLE
Credentials Rotation Playbooks

### PR DESCRIPTION
**Pull request by melamedbn**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4954)

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 1]
```diff
@@ -0,0 +1,805 @@
...
+
+  - **New Key Generation**: After ensuring the old key is disabled, generate a new access key.
+
+  ### **GSuite Admin Remediation**
+  Admin accounts are crucial:
+  - **Reset Password**: Resets the user password to halt any unauthorized access.
+
+  - **Revoke Access Token**: Revoke any suspicious or unauthorized access tokens.
+
+  - **Combo Action**: Both resetting the password and revoking access tokens ensure complete safety.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:19]:
```suggestion
  - **Combo Action**: Reset the password and revoke access tokens to ensure complete safety.
```

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 2]
```diff
@@ -0,0 +1,805 @@
...
+    isautoswitchedtoquietmode: false
+  "26":
+    id: "26"
+    taskid: f1fec12d-af5e-4f93-8a4b-5285fad4314a
+    type: regular
+    task:
+      id: f1fec12d-af5e-4f93-8a4b-5285fad4314a
+      version: -1
+      name: Generate Password
+      description: "This function generates a password and allows various parameters to customize the properties of the password depending on the use case (e.g. password complexity requirements).  The default behavior is to generate a password of  *random length* including all four character classes (upper, lower, digits, symbols) with at least five and at most ten characters per class. \n\nThe min_* values all default to 0. This means that if the command is executed in this way:\n!GeneratePassword max_lcase=10\nIt is possible that a password of length zero could be generated. It is therefore recommended to always include a min_* parameter that matches. \n\nThe debug parameter will print certain properties of the command into the WarRoom for easy diagnostics."
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:597]:
```suggestion
      description: "This function generates a password and allows various parameters to customize the properties of the password depending on the use case (e.g., password complexity requirements). The default behavior is to generate a password of  *random length* including all four character classes (upper, lower, digits, symbols) with at least five and at most ten characters per class. \n\nThe min_* values all default to 0. This means that if the command is executed in this way:\n!GeneratePassword max_lcase=10\nit is possible that a password of length zero could be generated. It is therefore recommended to always include a min_* parameter that matches. \n\nThe debug parameter will print certain properties of the command into the War Room for easy diagnostics."
```

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 3]
```diff
@@ -0,0 +1,805 @@
...
+- key: serviceAccountEmail
+  value: {}
+  required: false
+  description: The service account email.
+  playbookInputQuery:
+- key: identityType
+  value: {}
+  required: false
+  description: |-
+    The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:798]:
```suggestion
    The type of identity involved. Usually mapped to the incident field named 'cloudidentitytype'.
```

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 4]
```diff
@@ -0,0 +1,805 @@
...
+    e.g.
+    USER, SERVICE_ACCOUNT,APPLICATION
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:800]:
```suggestion
    e.g.,
    USER, SERVICE_ACCOUNT,APPLICATION
```
should there be a space in between entities?
After USER, there is a space
After SERVICE_ACCOUNT, there is no space

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+- **New Key Generation**: After ensuring the old key is disabled, generate a new access key.
+
+### **GSuite Admin Remediation**
+Admin accounts are crucial:
+- **Reset Password**: Resets the user password to halt any unauthorized access.
+
+- **Revoke Access Token**: Revoke any suspicious or unauthorized access tokens.
+
+- **Combo Action**: Both resetting the password and revoking access tokens ensure complete safety.
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md:15]:
```suggestion
- **Combo Action**: Both resetting the password and revoking access tokens to ensure complete safety.
```

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+This playbook uses the following sub-playbooks, integrations, and scripts.
+
+### Sub-playbooks
+
+This playbook does not use any sub-playbooks.
+
+### Integrations
+
+* GSuiteAdmin
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md:27]:
```suggestion
GSuiteAdmin
```

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| GSuiteRemediationType | The response playbook provides the following remediation actions using GSuite Admin:<br/><br/>Reset: By entering "Reset" in the input, the playbook will execute password reset.<br/><br/>Revoke: By entering "Revoke" in the input, the playbook will execute access key suspension. \(Both the user and client IDs must be provided\)<br/><br/>ALL: By entering "ALL" in the input, the playbook will execute the password reset and revoke access key tasks. |  | Optional |
+| userID | Identifies the user in the API request. The value can be the user's primary email address, alias email address, or unique user ID. |  | Optional |
+| clientID | The client ID. |  | Optional |
+| zone | The name of the zone. |  | Optional |
+| serviceAccountEmail | The service account email. |  | Optional |
+| identityType | The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.<br/>e.g.<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP_README.md:55]:
```suggestion
| identityType | The type of identity involved. Usually mapped to the incident field named 'cloudidentitytype'.<br/>e.g.,<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
```
should there be a space in between entities?
After USER, there is a space
After SERVICE_ACCOUNT, there is no space

File: Packs/GCP-Enrichment-Remediation/ReleaseNotes/1_1_9.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Playbooks
+
+##### New: Cloud Credentials Rotation - GCP
```
Comment (ShirleyDenkberg) [inline Packs/GCP-Enrichment-Remediation/ReleaseNotes/1_1_9.md:4]: Need a description of the playbook.

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| IAMRemediationType | The response playbook provides the following remediation actions for IAM users:<br/><br/>Reset - By entering "Reset" in the input, the playbook will execute password reset.<br/><br/>Deactivate - By entering "Deactivate" in the input, the playbook will execute access key deactivation.<br/><br/>ALL - By entering "Both" in the input, the playbook will execute both password reset and access key deactivation. |  | Optional |
+| shouldCloneSA | Whether to clone the compromised SA before putting a deny policy to it. |  | Optional |
+| identityType | The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.<br/>e.g.<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md:60]: Will fix that, thanks.

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+* aws-iam-get-instance-profile
+* aws-iam-list-role-policies
+
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| IAMRemediationType | The response playbook provides the following remediation actions for IAM users:<br/><br/>Reset - By entering "Reset" in the input, the playbook will execute password reset.<br/><br/>Deactivate - By entering "Deactivate" in the input, the playbook will execute access key deactivation.<br/><br/>ALL - By entering "Both" in the input, the playbook will execute both password reset and access key deactivation. |  | Optional |
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md:58]: Fixed.

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+| shouldCloneSA | Whether to clone the compromised SA before putting a deny policy to it. |  | Optional |
+| identityType | The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.<br/>e.g.<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
+| newRoleName | The new role name to assign in the clone service account flow. |  | Optional |
+| newInstanceProfileName | The new instance profile name to assign in the clone service account flow. |  | Optional |
+| accessKeyID | The access key Id. |  | Optional |
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS_README.md:63]: Fixed

File: Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_6.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+
+#### Playbooks
+
+##### New: Cloud Credentials Rotation - AWS
+
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/ReleaseNotes/1_1_6.md:5]: Added.

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
+## **Azure Credentials Rotation Playbook**
+
+### **IAM Remediation**
+Protect your identity and access management:
+- **Reset Password**: Resets the user password to halt any unauthorized access.
+
+- **Revoke Session**: Terminate current active sessions to ensure the malicious actor is locked out.
+
+- **Combo Action**: Consider both resetting the password and revoking all active sessions.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md:9]: Fixed.

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+## Playbook Inputs
+
+---
+
+| **Name** | **Description** | **Default Value** | **Required** |
+| --- | --- | --- | --- |
+| IAMRemediationType | The response playbook provides the following remediation actions using MSGraph Users:<br/><br/>Reset: By entering "Reset" in the input, the playbook will execute password reset.<br/><br/>Revoke: By entering "Revoke" in the input, the playbook will revoke the user's session.<br/><br/>ALL: By entering "ALL" in the input, the playbook will execute the reset password and revoke session tasks. |  | Optional |
+| appID | This is the unique application \(client\) ID of the application. |  | Optional |
+| objectID | This is the unique ID of the service principal object associated with the application. |  | Optional |
+| userID | The user Id or user principal name. |  | Optional |
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md:49]: Fixed

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+| identityType | The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.<br/>e.g.<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure_README.md:50]: Fixed.

File: Packs/Azure-Enrichment-Remediation/ReleaseNotes/1_1_7.md [hunk 16]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+
+##### New: Cloud Credentials Rotation - Azure
+
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/ReleaseNotes/1_1_7.md:5]: fixed

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml [hunk 17]
```diff
@@ -0,0 +1,643 @@
...
+  playbookInputQuery:
+- key: identityType
+  value: {}
+  required: false
+  description: |-
+    The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.
+    e.g.
+    USER,SERVICE_ACCOUNT,APPLICATION
+  playbookInputQuery:
+outputs: []
```
Comment (ssokolovich) [inline Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml:640]: no outputs?

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 18]
```diff
@@ -0,0 +1,805 @@
...
+      version: -1
+      name: List service accounts
+      description: 'Lists service accounts in project, or retrieves a specific service accounts information. One of the arguments: ''service_account_name'' or ''project_name''  must be provided.'
+      script: '|||gcp-iam-service-accounts-get'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "25"
```
Comment (ssokolovich) [inline Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:645]: may I suggest adding a conditional task that will check if the required service account was returned? cause if the account isn't there the PB might fail later on ( as you are using filters and transformers in the conditions).

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 19]
```diff
@@ -0,0 +1,1485 @@
...
+    isautoswitchedtoquietmode: false
+  "31":
+    id: "31"
+    taskid: 658a17ae-262d-40e4-891e-2ea2f8760cb7
+    type: regular
+    task:
+      id: 658a17ae-262d-40e4-891e-2ea2f8760cb7
+      version: -1
+      name: Attach role policy
+      description: Attaches the specified managed policy to the specified IAM Entity.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:568]: Fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 20]
```diff
@@ -0,0 +1,1485 @@
...
+- key: shouldCloneSA
+  value: {}
+  required: false
+  description: Whether to clone the compromised SA before putting a deny policy to it.
+  playbookInputQuery:
+- key: identityType
+  value: {}
+  required: false
+  description: |-
+    The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1448]: Fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 21]
```diff
@@ -0,0 +1,1485 @@
...
+  "16":
+    id: "16"
+    taskid: b1c6f9b4-50f4-418d-8f61-b09ce8bd59df
+    type: regular
+    task:
+      id: b1c6f9b4-50f4-418d-8f61-b09ce8bd59df
+      version: -1
+      name: Deactivate the user's access key
+      description: Changes the status of the specified access key from Active to Inactive, or vice versa. This operation can be used to disable a user's key as part of a key rotation workflow.
+      script: AWS - IAM|||aws-iam-update-access-key
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:123]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 22]
```diff
@@ -0,0 +1,1485 @@
...
+  "17":
+    id: "17"
+    taskid: 816121ba-9b51-48f0-8bfd-cb360284a4d1
+    type: regular
+    task:
+      id: 816121ba-9b51-48f0-8bfd-cb360284a4d1
+      version: -1
+      name: Force password reset
+      description: Changes the password for the specified IAM user.
+      script: AWS - IAM|||aws-iam-update-login-profile
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:161]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 23]
```diff
@@ -0,0 +1,1485 @@
...
+    nexttasks:
+      '#default#':
+      - "30"
+      "Yes":
+      - "29"
+    separatecontext: false
+    conditions:
+    - label: "Yes"
+      condition:
+      - - operator: isEqualString
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:690]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 24]
```diff
@@ -0,0 +1,1485 @@
...
+  "41":
+    id: "41"
+    taskid: 22f43dcc-ecaa-444c-8e9c-d1d461e0099c
+    type: regular
+    task:
+      id: 22f43dcc-ecaa-444c-8e9c-d1d461e0099c
+      version: -1
+      name: Set attached policies ARN
+      description: Set a value in context under the key you entered.
+      scriptName: Set
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:882]: Added! Great feedback!

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 25]
```diff
@@ -0,0 +1,1485 @@
...
+  "42":
+    id: "42"
+    taskid: 2fda800e-a38f-47aa-8948-faaf2abe2200
+    type: regular
+    task:
+      id: 2fda800e-a38f-47aa-8948-faaf2abe2200
+      version: -1
+      name: Set policies name
+      description: Set a value in context under the key you entered.
+      scriptName: Set
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:921]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 26]
```diff
@@ -0,0 +1,1485 @@
...
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "43"
+    scriptarguments:
+      policyName:
+        complex:
+          root: policyNames
+      roleName:
+        simple: AmazonEKSNodeRole
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1020]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 27]
```diff
@@ -0,0 +1,1485 @@
...
+      version: -1
+      name: Put role policy
+      description: Adds or updates an inline policy document that is embedded in the specified IAM role.
+      script: '|||aws-iam-put-role-policy'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "36"
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1199]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 28]
```diff
@@ -0,0 +1,1485 @@
...
+      - "21"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: isEmpty
+          left:
+            value:
+              complex:
+                root: inputs.roleName
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1381]: fixed

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 29]
```diff
@@ -0,0 +1,1485 @@
...
+    Reset - By entering "Reset" in the input, the playbook will execute password reset.
+
+    Deactivate - By entering "Deactivate" in the input, the playbook will execute access key deactivation.
+
+    ALL - By entering "ALL" in the input, the playbook will execute both password reset and access key deactivation.
+  playbookInputQuery:
+- key: shouldCloneSA
+  value: {}
+  required: false
+  description: Whether to clone the compromised SA before putting a deny policy to it.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1442]: added

File: Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml [hunk 30]
```diff
@@ -0,0 +1,1485 @@
...
+  playbookInputQuery:
+- key: identityType
+  value: {}
+  required: false
+  description: |-
+    The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.
+    e.g.
+    USER,SERVICE_ACCOUNT,APPLICATION
+  playbookInputQuery:
+- key: newRoleName
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/AWS-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_AWS.yml:1452]: added

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml [hunk 31]
```diff
@@ -0,0 +1,643 @@
...
+      condition:
+      - - operator: isEqualString
+          left:
+            value:
+              complex:
+                root: inputs.identityType
+            iscontext: true
+          right:
+            value:
+              simple: IAM
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml:527]: fixed

File: Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml (hunk 17 above)
Comment (melamedbn) [reply Packs/Azure-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_Azure.yml:640]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 32]
```diff
@@ -0,0 +1,502 @@
+id: Cloud Credentials Rotation - Generic
+version: -1
+name: Cloud Credentials Rotation - Generic
+description: |-
+  ## **Cloud Credentials Rotation - Generic**
+
+  This comprehensive playbook combine the remediation steps from AWS, Azure, and GCP sub-playbooks into a single, cohesive guide. Regardless of which Cloud Service Provider (CSP) you're working with, this playbook will direct you to the relevant steps, ensuring swift and effective response.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:7]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 33]
```diff
@@ -0,0 +1,502 @@
...
+      description: |-
+        ## **Azure Credentials Rotation Playbook**
+
+        ### **IAM Remediation**
+        Protect your identity and access management:
+        - **Reset Password**: Resets the user password to halt any unauthorized access.
+
+        - **Revoke Session**: Terminate current active sessions to ensure the malicious actor is locked out.
+
+        - **Combo Action**: Consider both resetting the password and revoking all active sessions.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:75]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 34]
```diff
@@ -0,0 +1,502 @@
...
+
+        - **New Key Generation**: After ensuring the old key is disabled, generate a new access key.
+
+        ### **GSuite Admin Remediation**
+        Admin accounts are crucial:
+        - **Reset Password**: Resets the user password to halt any unauthorized access.
+
+        - **Revoke Access Token**: Revoke any suspicious or unauthorized access tokens.
+
+        - **Combo Action**: Both resetting the password and revoking access tokens ensure complete safety.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:147]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 35]
```diff
@@ -0,0 +1,502 @@
...
+      instanceID:
+        complex:
+          root: inputs.AWS-instanceID
+      newInstanceProfileName:
+        complex:
+          root: inputs.AWS-newInstanceProfileName
+      newRoleName:
+        complex:
+          root: inputs.AWS-newRoleName
+      roleName:
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:273]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 36]
```diff
@@ -0,0 +1,502 @@
...
+    type: condition
+    task:
+      id: 5fd2119c-3fd8-4534-8e8c-8ad72738bd5d
+      version: -1
+      name: Select cloud provider
+      description: Checks the cloud provider.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:315]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 37]
```diff
@@ -0,0 +1,502 @@
...
+    Revoke: By entering "Revoke" in the input, the GCP will revoke the access key, GSuite Admin will revoke the access token and the MSGraph Users will revoke the session.
+    Supports: GCP, GSuite Admin and MSGraph Users.
+
+    Deactivate - By entering "Deactivate" in the input, the playbook will execute access key deactivation.
+    Supports: AWS.
+
+    ALL: By entering "ALL" in the input, the playbook will execute the all remediation actions provided for each CSP.
+  playbookInputQuery:
+- key: shouldCloneSA
+  value: {}
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:407]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 38]
```diff
@@ -0,0 +1,502 @@
...
+  required: false
+  description: The client ID.
+  playbookInputQuery:
+- key: GCP-zone
+  value: {}
+  required: false
+  description: |-
+    The name of the zone.
+    e.g.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:428]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 39]
```diff
@@ -0,0 +1,502 @@
...
+  playbookInputQuery:
+- key: Azure-ObjectID
+  value: {}
+  required: false
+  description: This is the unique ID of the service principal object associated with the application.
+  playbookInputQuery:
+- key: Azure-userID
+  value: {}
+  required: false
+  description: The user Id or user principal name.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:450]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 40]
```diff
@@ -0,0 +1,502 @@
...
+  playbookInputQuery:
+- key: AWS-userID
+  value: {}
+  required: false
+  description: The user name.
+  playbookInputQuery:
+- key: AWS-accessKeyID
+  value: {}
+  required: false
+  description: The access key Id.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:465]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml [hunk 41]
```diff
@@ -0,0 +1,502 @@
...
+    e.g.
+    AWS, AZURE, GCP
+  playbookInputQuery:
+- key: identityType
+  value: {}
+  required: false
+  description: |-
+    The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.
+    e.g.
+    USER, SERVICE_ACCOUNT,APPLICATION
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic.yml:496]: fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic_README.md [hunk 42]
```diff
@@ -0,0 +1,78 @@
...
+| AWS-newRoleName | The name of the new role to create if the analyst decides to clone the service account. |  | Optional |
+| AWS-newInstanceProfileName | The name of the new instance profile to create if the analyst decides to clone the service account. |  | Optional |
+| AWS-roleNameToRestrict | If provided, the role will be attached with a deny policy without the compute instance analysis flow. |  | Optional |
+| cloudProvider | The CSP that triggered the alert. Usually mapped to incident field named 'cloudprovider'.<br/>e.g.,<br/>AWS, AZURE, GCP |  | Optional |
+| identityType | The type of identity involved. Usually mapped to incident field named 'cloudidentitytype'.<br/>e.g.,<br/>USER, SERVICE_ACCOUNT,APPLICATION |  | Optional |
+
+## Playbook Outputs
+
+---
+There are no outputs for this playbook.
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/CommonPlaybooks/Playbooks/playbook-Cloud_Credentials_Rotation_-_Generic_README.md:72]: will be fixed

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml [hunk 43]
```diff
@@ -0,0 +1,805 @@
...
+    task:
+      id: 9fe550ee-a468-4bc7-833a-17e3c15a8186
+      version: -1
+      name: Check remediation type
+      description: Checks the remediation type provided by the user.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      ALL:
```
[the remark this answers is not in the document]
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:338]: added

File: Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml (hunk 18 above)
Comment (melamedbn) [reply Packs/GCP-Enrichment-Remediation/Playbooks/playbook-Cloud_Credentials_Rotation_-_GCP.yml:645]: added

**Review by altmannyarden — APPROVED** [review]

**Pull request merged**